JSDoc comment string with the keyword "@private" before import statement in JS file result in cryptic error TS1191 during compilation

### 🔎 Search Terms

comment import statement bug private

### 🕗 Version & Regression Information

Version 5.4.5

I am the author and maintainer of the `ejs` library, which is widely used and has to work in old JavaScript runtimes. I am exploring `tsc` as a possibility for writing more modern JS, but transpiling back to ES5 for release. I have a tsconfig with `allowJs` true, and a single JS file where the actual code is a single import statement, and running `tsc` on it resulted in a cryptic error message, ` TS1191: An import declaration cannot have modifiers.`

After more than an hour of fruitlessly searching the Web, and trying different things, I finally realized that this error was happening because of a multi-line comments (specifically JSDoc), just above my import statement that contained the string `@private`.





### ⏯ Playground Link

_No response_

### 💻 Code

Minimal breaking example. Actual JSDoc in the comment is much more verbose, but does include `@private` as a reference to the namespace

```js
'use strict';

/** @private */

import { foo } from './test.js';

console.log(foo);
```


### 🙁 Actual behavior

Compiler exits with `error TS1191: An import declaration cannot have modifiers.`

### 🙂 Expected behavior

I expected it to compile my file, because my file is valid JavaScript.

### Additional information about the issue

_No response_

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -46279,7 +46279,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             // If we hit an import declaration in an illegal context, just bail out to avoid cascading errors.
             return;
         }
-        if (!checkGrammarModifiers(node) && hasEffectiveModifiers(node)) {
+        if (!checkGrammarModifiers(node) && node.modifiers) {
             grammarErrorOnFirstToken(node, Diagnostics.An_import_declaration_cannot_have_modifiers);
         }
         if (checkExternalImportOrExportDeclaration(node)) {

--- a/tests/baselines/reference/es6ImportWithJsDocTags.symbols
+++ b/tests/baselines/reference/es6ImportWithJsDocTags.symbols
@@ -1,0 +1,20 @@
+//// [tests/cases/compiler/es6ImportWithJsDocTags.ts] ////
+
+=== ./a.js ===
+export const foo = 1;
+>foo : Symbol(foo, Decl(a.js, 0, 12))
+
+=== ./b.js ===
+'use strict';
+
+/** @private */
+
+import { foo } from './a.js';
+>foo : Symbol(foo, Decl(b.js, 4, 8))
+
+console.log(foo);
+>console.log : Symbol(Console.log, Decl(lib.dom.d.ts, --, --))
+>console : Symbol(console, Decl(lib.dom.d.ts, --, --))
+>log : Symbol(Console.log, Decl(lib.dom.d.ts, --, --))
+>foo : Symbol(foo, Decl(b.js, 4, 8))
+

--- a/tests/baselines/reference/es6ImportWithJsDocTags.types
+++ b/tests/baselines/reference/es6ImportWithJsDocTags.types
@@ -1,0 +1,32 @@
+//// [tests/cases/compiler/es6ImportWithJsDocTags.ts] ////
+
+=== ./a.js ===
+export const foo = 1;
+>foo : 1
+>    : ^
+>1 : 1
+>  : ^
+
+=== ./b.js ===
+'use strict';
+>'use strict' : "use strict"
+>             : ^^^^^^^^^^^^
+
+/** @private */
+
+import { foo } from './a.js';
+>foo : 1
+>    : ^
+
+console.log(foo);
+>console.log(foo) : void
+>                 : ^^^^
+>console.log : (...data: any[]) => void
+>            : ^^^^    ^^     ^^^^^^^^^
+>console : Console
+>        : ^^^^^^^
+>log : (...data: any[]) => void
+>    : ^^^^    ^^     ^^^^^^^^^
+>foo : 1
+>    : ^
+

--- a/tests/cases/compiler/es6ImportWithJsDocTags.ts
+++ b/tests/cases/compiler/es6ImportWithJsDocTags.ts
@@ -1,0 +1,14 @@
+// @allowJs: true
+// @checkJs: true
+// @noEmit: true
+// @filename: ./a.js
+export const foo = 1;
+
+// @filename: ./b.js
+'use strict';
+
+/** @private */
+
+import { foo } from './a.js';
+
+console.log(foo);

